Public/Src/Cache/ContentStore/Library/Stores/FileSystemContentStoreInternal.cs

```diff
@@ -611,23 +611,22 @@ private void DeleteTempFolder()
 
             try
             {
-                var contentHashes = ReadContentHashesFromDisk(context);
-                _tracer.Debug(context, $"Enumerated {contentHashes.Count} entries by {stopwatch.ElapsedMilliseconds}ms.");
+                var contentHashes = ReadSnapshotFromDisk(context);
+                _tracer.Debug(context, $"Enumerated {contentHashes.Count} entries in {stopwatch.ElapsedMilliseconds}ms.");
 
-                var hashInfoPairs = new List<KeyValuePair<ContentHash, ContentFileInfo>>();
-                foreach (var grouping in contentHashes.GroupBy(hash => hash))
+                // We are using a list of classes instead of structs due to the maximum object size restriction

```

Comment:
This is very minor, but I think it make sense to move this grouping logic into a factory method inside ContentDirectorySnapshot? In this case it'll have more "behavior" and stop being just plain data, IMO.

Reply:
This is the only usage of the method, it seems a little bit overkill to make this change for now, as it would not be saving us from duplicate code.

---
In reply to: [285689134](https://github.com/microsoft/BuildXL/pull/329#discussion_r285689134) [](ancestors = 285689134)